crates/ide-db/src/search.rs

```diff
@@ -503,6 +503,20 @@ impl<'a> FindUsages<'a> {
                 if !search_range.contains_inclusive(offset) {
                     return None;
                 }
+                // If this is not a word boundary, that means this is only part of an identifier,
+                // so it can't be what we're looking for.
+                // This speeds up short identifiers significantly.
+                if text[..idx]
+                    .chars()
+                    .next_back()
+                    .is_some_and(|ch| matches!(ch, 'A'..='Z' | 'a'..='z' | '_'))
```

Comment:
To clarify for myself, we are effectively checking if we are searching for some string of length n, but our match is an identifier that is longer than n?

Note that this predicate will break searches for unicode identifiers

Reply:
> To clarify for myself, we are effectively checking if we are searching for some string of length n, but our match is an identifier that is longer than n?

Yes, exactly.

> Note that this predicate will break searches for unicode identifiers

No? I use `chars()`.

Reply:
I just realized unicode just doesn't benefit from this but its not like it matters in that case, sorry